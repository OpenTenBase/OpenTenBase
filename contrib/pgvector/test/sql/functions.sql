SET client_min_messages = warning;
CREATE EXTENSION IF NOT EXISTS vector;

SELECT '[1,2,3]'::vector + '[4,5,6]';
SELECT '[1,2,3]'::vector - '[4,5,6]';

SELECT vector_dims('[1,2,3]');
SELECT round(vector_norm('[1,1]')::numeric, 5);

SELECT round(l2_distance('[1,2]', '[0,0]')::numeric, 5);
SELECT l2_distance('[1,2]', '[3]');

SELECT inner_product('[1,2]', '[3,4]');
SELECT inner_product('[1,2]', '[3]');

SELECT round(cosine_distance('[1,2]', '[2,4]')::numeric, 5);
SELECT cosine_distance('[1,2]', '[0,0]');
SELECT cosine_distance('[1,2]', '[3]');
