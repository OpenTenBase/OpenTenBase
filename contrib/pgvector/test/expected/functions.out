SET client_min_messages = warning;
CREATE EXTENSION IF NOT EXISTS vector;
SELECT '[1,2,3]'::vector + '[4,5,6]';
 ?column? 
----------
 [5,7,9]
(1 row)

SELECT '[1,2,3]'::vector - '[4,5,6]';
  ?column?  
------------
 [-3,-3,-3]
(1 row)

SELECT vector_dims('[1,2,3]');
 vector_dims 
-------------
           3
(1 row)

SELECT round(vector_norm('[1,1]')::numeric, 5);
  round  
---------
 1.41421
(1 row)

SELECT round(l2_distance('[1,2]', '[0,0]')::numeric, 5);
  round  
---------
 2.23607
(1 row)

SELECT l2_distance('[1,2]', '[3]');
ERROR:  different vector dimensions 2 and 1
SELECT inner_product('[1,2]', '[3,4]');
 inner_product 
---------------
            11
(1 row)

SELECT inner_product('[1,2]', '[3]');
ERROR:  different vector dimensions 2 and 1
SELECT round(cosine_distance('[1,2]', '[2,4]')::numeric, 5);
  round  
---------
 0.00000
(1 row)

SELECT cosine_distance('[1,2]', '[0,0]');
 cosine_distance 
-----------------
             NaN
(1 row)

SELECT cosine_distance('[1,2]', '[3]');
ERROR:  different vector dimensions 2 and 1
